docs/developer-resources.md

```diff
@@ -0,0 +1,346 @@
+
+##### com.monumental.controllers
+
+These are standard issue Spring Boot `RestController` classes. They use `ModelService` classes primarily to do the business logic, but for small things may sometimes directly call `JpaRepository` classes. If a controller method is more than 5 lines long, it probably can be moved into the appropriate `ModelService` class (we're not always great at following this rule).
+
+##### com.monumental.security
+
+This is a special package with a few key classes that configure Spring Boot Security to work with our `User` model and configure it to meet our needs. You shouldn't really need to touch this package too much, but it's important to know that:
+
+1. `JpaAuditConfiguration#getCurrentAuditor` is responsible for finding the logged in user's `User` row in the database
+2. `ProductionSecurityConfig` is responsible for switching on HTTPS in production (so that you don't need to set up SSL locally)
+3. `SecurityConfig` is responsible for setting up `/api/login` and `/api/logout`, as well as making most the api endpoints **un**authenticated because the endpoints then define their own authentication using `@PreAuthorize`
+4. `UserAwareUserDetails` is our custom session object that holds a reference to the appropriate `User` row returned by `JpaAuditConfiguration#getCurrentAuditor`.
+5. `UserDetailsServiceImpl` is responsible for implementing `/api/login`.
+
+##### com.monumental.util
+
+This is kind of the garbage dump of everything that doesn't fit in one of the above packages. It has some useful utils for basic Java things as well as some more complex things, such as `AsyncJob`s for handling long running requests in a separate thread (such as for bulk operations).
+
+### Client Tier
+
+The client is a React app created with `create-react-app`. Note that this app has **not** been ejected, which means you still must use `react-scripts` to build and run it. I would strongly advise against ever ejecting the project. It should not be necessary as we use a very standard React setup.
+
+#### Folders
+
+We categorize our React components by three criteria:
+
+1. Is it stateful - Does it keep state, is it connected to Redux, is it aware of React Router, etc.
+2. Is it presentational - Is it concerned about the way things look to the user.
+3. Is it a page, or not a page - We make this distinction because it helps keep things organized.
+
+##### pages
+
+This folder contains the components that server as page entry points. These components are **stateful** and **non-presentational**. They render components from the `components` and `containers` folders. These components should generally only be rendered by `<Route>`s in `App.js` but there are a couple key exceptions, such as on the admin panel where we render the `SearchPage` within the `AdminPage`.
```

Comment:
`serve as page entry points`, not `server as page entry points`

Reply:
I got servers on the brain.